### PR TITLE
Drop do-nothing ee class CatalogDelegate.

### PR DESCRIPTION
**Pull request by paulmartel**
Also, in TableCatalogDelegate skip needless passing of data members into member functions.

### CODE REVIEW

**Comment by jxchinaren** [comment]
@paulmartel Looks good to me.

File: src/ee/storage/TableCatalogDelegate.hpp [hunk 1]
```diff
@@ -103,48 +105,32 @@ class TableCatalogDelegate : public CatalogDelegate {
...
 
-    void setTable(Table * tb) {
-        m_table = tb;
-    }
+    void setTable(Table * tb) { m_table = tb; }
 
-    bool exportEnabled() {
-        return m_exportEnabled;
-    }
+    bool exportEnabled() { return m_exportEnabled; }
```
Comment (hhakimi53) [inline src/ee/storage/TableCatalogDelegate.hpp]: May be can make this function also const

bool exportEnabled() const { return m_exportEnabled; }

File: src/ee/storage/TableCatalogDelegate.hpp [hunk 2]
```diff
@@ -103,48 +105,32 @@ class TableCatalogDelegate : public CatalogDelegate {
...
 
-    std::string signature() {
-        return m_signature;
-    }
+    const std::string& signature() { return m_signature; }
```
Comment (hhakimi53) [inline src/ee/storage/TableCatalogDelegate.hpp]: Same suggestion for this function also.

const std::string& signature() const { return m_signature; }

File: src/ee/storage/TableCatalogDelegate.hpp [hunk 3]
```diff
@@ -103,48 +105,32 @@ class TableCatalogDelegate : public CatalogDelegate {
...
 
-    const char* signatureHash() {
-        return m_signatureHash;
-    }
+    const char* signatureHash() { return m_signatureHash; }
```
Comment (hhakimi53) [inline src/ee/storage/TableCatalogDelegate.hpp]: Same suggestion for this function also.

File: src/ee/storage/TableCatalogDelegate.hpp [hunk 4]
```diff
@@ -103,48 +105,32 @@ class TableCatalogDelegate : public CatalogDelegate {
...
 
     /*
      * Returns true if this table is a materialized view
      */
-    bool materialized() {
-        return m_materialized;
-    }
+    bool materialized() { return m_materialized; }
```
Comment (hhakimi53) [inline src/ee/storage/TableCatalogDelegate.hpp]: Same suggestion for this function also.

Comment (hhakimi53) [inline src/ee/storage/TableCatalogDelegate.hpp]: Overall changes look good.

**Pull request closed without merging**